### ISSUE TITLE
[A11ySev2] [Programmatic access -Dashboards with Grafana - Select data source dialog]: Unnecessary h2 heading defined for list items in Select data source dialog.

### ISSUE THREAD

**Issue by LinglingTong** [grafana/grafana#118779]
## Steps to reproduce

Repro Steps: 
  
 
1. Open 'Azure Monitor dashboards with Grafana' in Azure Portal
2. "Monitor | Dashboards with Grafana | Gallery" screen appears with already Dashboards with Grafana tab selected.
3. Navigate to the all the controls available on Dashboards with Grafana blade like tool bar , controls under All Dashboards tab.
4. Navigate New menu present in the tool bar and select New Dashboard.
5. Navigate to "Add visualization" button and hit ENTER key. "Select data source" popup will appear.
6. Navigate to  all the control present in the Select data source dialog.
7. Observe that screen reader announcement for second radio button under the refresh combo box. 

**Actual Result:**
Each data source list item is announced as a level 2 heading. 
Screen readers announce repeated “Heading level 2” for every item in the list. 

**Expected Result:**
List items in the Select data source dialog should not be marked as headings. 
Appropriate semantic elements (e.g., li, button, or role="option") should be used instead. 
Screen readers should announce each item as a selectable option without heading level information, providing a clear and efficient navigation experience.

**Relevant WCAG Criteria:** [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html) (Level A)

## Environment

- **Grafana version:** 12.2.0
- **Data source type & version:**
- **User OS & Browser:** Version 25H2 (OS Build 26200.7093) / Chrome Version 142.0.7444.177 (Official Build) (64-bit)
- **Others:**

**Comment by LinglingTong** [comment]
<img width="979" height="584" alt="Image" src="https://github.com/user-attachments/assets/ef4aaf75-8b11-45fe-b254-819d7290401f" />
